echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/PipelineCache.java

```diff
@@ -24,70 +25,98 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import java.util.stream.Collectors;
 
 import static java.time.Instant.now;
-import static java.util.concurrent.TimeUnit.SECONDS;
 
 @Component
 @Slf4j
 public class PipelineCache implements MonitoredPoller {
-  private final Scheduler scheduler;
   private final int pollingIntervalSeconds;
   private final Front50Service front50;
   private final Registry registry;
+  private final ScheduledExecutorService executorService;
 
+  private transient Boolean running;
   private transient Instant lastPollTimestamp;
-  private transient Subscription subscription;
 
-  private transient SerializedSubject<List<Pipeline>, List<Pipeline>> pipelineSubject = ReplaySubject.<List<Pipeline>>createWithSize(1).toSerialized();
+  @Nullable
+  private transient List<Pipeline> pipelines;
 
   @Autowired
-  public PipelineCache(@NonNull Scheduler scheduler,
+  public PipelineCache(int pollingIntervalSeconds,
+                       @NonNull Front50Service front50,
+                       @NonNull Registry registry) {
+    this(Executors.newSingleThreadScheduledExecutor(), pollingIntervalSeconds, front50, registry);
+  }
+
+  // VisibleForTesting
+  public PipelineCache(ScheduledExecutorService executorService,
                        int pollingIntervalSeconds,
                        @NonNull Front50Service front50,
                        @NonNull Registry registry) {
-    this.scheduler = scheduler;
+    this.executorService = executorService;
     this.pollingIntervalSeconds = pollingIntervalSeconds;
     this.front50 = front50;
     this.registry = registry;
+    this.running = false;
+    this.pipelines = null;
+  }
+
+  @PreDestroy
+  public void stop() {
+    running = false;
   }
 
   @PostConstruct
   public void start() {
-    if (subscription == null || subscription.isUnsubscribed()) {
-      subscription = Observable.interval(pollingIntervalSeconds, SECONDS, scheduler)
-        .doOnNext(this::onFront50Request)
-        .flatMap(tick -> front50.getPipelines())
-        .doOnError(this::onFront50Error)
-        .retry()
-        .map(PipelineCache::decorateTriggers)
-        .doOnNext(this::logRefresh)
-        .subscribe(pipelineSubject);
-    }
+    running = true;
+
+    executorService.scheduleWithFixedDelay(
+      new Runnable() {
+        @Override
+        public void run() {
+          pollPipelineConfigs();
+        }
+      },
+      0, pollingIntervalSeconds, TimeUnit.SECONDS);
   }
 
-  @PreDestroy
-  public void stop() {
-    if (subscription != null) {
-      subscription.unsubscribe();
+  // VisibleForTesting
+  void pollPipelineConfigs() {
+    if (!isRunning()) {
+      return;
+    }
+
+    try {
+      log.debug("Getting pipelines from Front50...");
+      long start = System.currentTimeMillis();
+      pipelines = decorateTriggers(front50.getPipelines());
+
+      lastPollTimestamp = now();
+      registry.counter("front50.requests").increment();
```

Comment:
If we're tracking the timing of this call, why not just `registry.timer` it and change `front50.requests` to a timer?

Reply:
I guess we could, but if I'm going to change this I'd rather remove this kind of one-off metrics and rely on the http client to publish metrics. If echo picks up the okhttp client from kork it _should_ already be getting some standard metrics for free?